src/System.Runtime.InteropServices/tests/System/Runtime/InteropServices/Marshal/StringMarshalingTests.cs

```diff
@@ -0,0 +1,187 @@
+                        Assert.Equal(ts.Substring(0, ts.Length - 1), str2);
+                    }
+                }
+                finally
+                {
+                    Marshal.FreeCoTaskMem(UniStr);
+                }
+            }
+        }
+
+        [Fact]
+        public void StringToHGlobalAnsiToString()
+        {
+            foreach (String ts in TestStrings)
+            {
+                if (ts.Contains("\0"))
+                    continue; //Skip the string with nulls case
+
+                IntPtr AnsiStr = Marshal.StringToHGlobalAnsi(ts);
+                try
+                {
+                    String str = Marshal.PtrToStringAnsi(AnsiStr);
+
+                    Assert.Equal(ts, str);
+                    if (ts.Length > 0)
+                    {
+                        String str2 = Marshal.PtrToStringAnsi(AnsiStr, ts.Length - 1);
+                        Assert.Equal(ts.Substring(0, ts.Length - 1), str2);
+                    }
+                }
+                finally
+                {
+                    Marshal.FreeHGlobal(AnsiStr);
+                }
+            }
+        }
+
+        [Fact]
+        public void StringToHGlobalUniToString()
+        {
+            foreach (String ts in TestStrings)
+            {
+                if (ts.Contains("\0"))
+                    continue; //Skip the string with nulls case
+
+                IntPtr UniStr = Marshal.StringToHGlobalUni(ts);
+                try
+                {
+                    String str = Marshal.PtrToStringUni(UniStr);
+
+                    Assert.Equal(ts, str);
+                    if (ts.Length > 0)
+                    {
+                        String str2 = Marshal.PtrToStringUni(UniStr, ts.Length - 1);
+                        Assert.Equal(ts.Substring(0, ts.Length - 1), str2);
+                    }
+                }
+                finally
+                {
+                    Marshal.FreeHGlobal(UniStr);
+                }
+            }
+
+        }
+
+        [Fact]
+        public void TestUTF8String()
+        {
+            foreach (String srcString in TestStrings)
+            {
+                // we assume string null terminated
+                if (srcString.Contains("\0"))
+                    continue;
+
+                IntPtr ptrString = Marshal.StringToCoTaskMemUTF8(srcString);
+                try
+                {
+                    string retString = Marshal.PtrToStringUTF8(ptrString);
+
+                    Assert.Equal(srcString, retString);
+                    if (srcString.Length > 0)
+                    {
+                        string retString2 = Marshal.PtrToStringUTF8(ptrString, srcString.Length - 1);
+                        Assert.Equal(srcString.Substring(0, srcString.Length - 1), retString2);
+                    }
+                }
+                finally
+                {
+                    Marshal.FreeHGlobal(ptrString);
```

Comment:
FreeHGlobal? the memory is created by CoTaskMem.

Reply:
Fixed with `Marshal.FreeCoTaskMem(ptrString);`